torchgeo/datamodules/deepglobelandcover.py

```diff
@@ -63,26 +107,76 @@ def setup(self, stage: Optional[str] = None) -> None:
 
         Args:
             stage: stage to set up
+
+        .. versionchanged:: 0.4
+            Add functionality to randomly crop patches from a tile during
+            training and pad validation and test samples to next multiple of 32
         """
-        transforms = Compose([self.preprocess])
 
-        dataset = DeepGlobeLandCover(
-            split="train", transforms=transforms, **self.kwargs
+        def n_random_crop(sample: Dict[str, Any]) -> Dict[str, Any]:
+            """Construct 'num_patches_per_tile' random patches of input tile.
+
+            Args:
+                sample: contains image and mask tile from dataset
+
+            Returns:
+                stacked randomly cropped patches from input tile
+            """
+            images, masks = [], []
+            for i in range(self.num_patches_per_tile):
+                image, mask = self.rcrop(sample["image"], sample["mask"].float())
+                images.append(image.squeeze(0))
+                masks.append(mask.squeeze().long())
+
+            sample["image"] = torch.stack(images)
+            sample["mask"] = torch.stack(masks)
+            return sample
+
+        def pad_to(sample: Dict[str, Any]) -> Dict[str, Any]:
```

Comment:
Presumably these are going to be used in multiple datamodules? Can we move these to `torchgeo/datamodules/utils.py`? Or maybe they should go in `torchgeo/transforms/*`? I'm trying to avoid adding transforms that aren't specific to geospatial data, but I'm not sure if there's anything like this in other libraries to avoid the duplication.

Reply:
Yes, they will be used in all datamodules, same with the `pad_to` function. It feels like it is more of an util than a transform, because `self.rcop` is a Kornia sequential augmentation it is just that we want to apply it `n`times to a single tile.

Reply:
Alright, let's move it to utils then

Reply:
Do you have a preferred name, for the `pad_to` function when it is in utils. Currently, it is explicitly used for segmentation samples where architectures rely on the dimensions being a multiple of 32. Would you want it to be a bit more universal where it accepts a  `multiple_of` argument to specify 32 or any other integer?

Similar question about the naming the `n_random_crop` in utils.

Reply:
No preference. I'm fine with keeping it specific to 32 until we find a case where we need some other multiple. For now, let's name it starting with an underscore and not export it in `__init__.py`. That way we reserve the right to rename or change it later without changing backwards compatibility.

Reply:
For the `n_random_crop` function I am a bit stuck. We use this function in `torch.Compose` which only accepts a single argument, but we need to specify both the number of patches to take and the Kornia augmentation operation. Would I have to write a small class that takes `kornia_augmen` and `num_patches` as init argument and then during the forward just accepts a sample and transforms it?

Reply:
Yes, it would be a class just like all transforms

Reply:
actually, I just found that the `AugmentationSequential` module in transforms.py is what we need, except that we want to apply it to generate `n` patches, so should I adapt that class to accept a 'num_patches_per_tile' argument?

Reply:
`AugmentationSequential` looks identical to `Compose` other than handling of dicts, I think you still need to pass a list of classes to it just like `Compose`.

Reply:
@isaaccorley wrote that, maybe he can comment.

Reply:
my idea was to define `AugmentationSequential` and then pass that into the Compose, but it doesn't seem to work, so I will define a new torch transform class.

Reply:
I think I found a way now, not sure if you will like it (will push). I wanted to test it on the actual DeepGlobe dataset and found that the link and directory structure is outdated. Should I do a separate PR for that?

Reply:
Yeah, prob separate PR